services/server/src/handlers/availabilitySlot.ts

```diff
@@ -0,0 +1,235 @@
+import { bad, conflict, forbidden, notfound } from "@/lib/error";
+import { datetime, id, pagination } from "@/validation/utils";
+import { isTutor, isTutorManager, isUser } from "@litespace/auth";
+import { IAvailabilitySlot } from "@litespace/types";
+import {
+  availabilitySlots,
+  interviews,
+  knex,
+  lessons,
+} from "@litespace/models";
+import dayjs from "@/lib/dayjs";
+import { NextFunction, Request, Response } from "express";
+import safeRequest from "express-async-handler";
+import { INTERVIEW_DURATION, isIntersecting } from "@litespace/sol";
+import zod from "zod";
+import { isEmpty } from "lodash";
+
+const findPayload = zod.object({
+  userId: id,
+  after: datetime,
+  before: datetime,
+  pagination,
+});
+
+const setPayload = zod.object({
+  slots: zod.array(
+    zod
+      .object({
+        action: zod.literal("create"),
+        start: datetime,
+        end: datetime,
+      })
+      .or(
+        zod
+          .object({
+            action: zod.literal("update"),
+            id: id,
+            start: datetime.optional(),
+            end: datetime.optional(),
+          })
+          .or(
+            zod.object({
+              action: zod.literal("delete"),
+              id: id,
+            })
+          )
+      )
+  ),
+});
+
+type CreateAction = {
+  action: "create";
+  start: string;
+  end: string;
+};
+type UpdateAction = {
+  action: "update";
+  id: number;
+  start?: string;
+  end?: string;
+};
+type DeleteAction = {
+  action: "delete";
+  id: number;
+};
+
+async function find(req: Request, res: Response, next: NextFunction) {
+  const user = req.user;
+  if (!isUser(user)) return next(forbidden());
+
+  const { userId, after, before, pagination } = findPayload.parse(req.body);
```

Comment:
Let's use the query and not the body. 

```suggestion
  const { userId, after, before, pagination } = findPayload.parse(req.query);
```